t/api/15-users.t

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env perl
+# Copyright (C) 2017-2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+use Test::Most;
+
+use Date::Format;
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use Test::Mojo;
+use Test::Warnings ':report_warnings';
+use OpenQA::Test::Case;
+use OpenQA::Client;
+require OpenQA::Schema::Result::Jobs;
+
+OpenQA::Test::Case->new->init_data(fixtures_glob => '03-users.pl');
+
+subtest 'Non admin cannot delete users' => sub {
+    my $t = Test::Mojo->new('OpenQA::WebAPI');
+    my $app = $t->app;
+    $t->ua(
+        OpenQA::Client->new(apikey => 'LANCELOTKEY01', apisecret => 'MANYPEOPLEKNOW')->ioloop(Mojo::IOLoop->singleton));
+    $t->app($app);
+
+    $t->delete_ok("/api/v1/user/99904")->status_is(403, "delete user failed.");
+};
+
+subtest 'Admins can delete users' => sub {
+    my $t = Test::Mojo->new('OpenQA::WebAPI');
+    my $app = $t->app;
+    $t->ua(
+        OpenQA::Client->new(apikey => 'ARTHURKEY01', apisecret => 'EXCALIBUR')->ioloop(Mojo::IOLoop->singleton));
+    $t->app($app);
+
+    $t->delete_ok("/api/v1/user/99904")->status_is(200, "delete user successfully.");
+};
+
+subtest 'Fails on non existen user' => sub {
+    my $t = Test::Mojo->new('OpenQA::WebAPI');
+    my $app = $t->app;
+    $t->ua(
+        OpenQA::Client->new(apikey => 'ARTHURKEY01', apisecret => 'EXCALIBUR')->ioloop(Mojo::IOLoop->singleton));
+    $t->app($app);
+
+    $t->delete_ok("/api/v1/user/99999")->status_is(404, "Non existent user.");
+};
```

Comment:
This looks rather expensive and duplicating. Can you just update the client in the existing webAPI app instance rather than recreating the app?

Please also remove the dot at end of line within the test messages, e.g. just

```perl
$t->delete_ok('/api/v1/user/99999')->status_is(404, 'Non existent user');
```

Reply:
You are right, I simply this and reorganized the code to be more efficient.

Reply:
new commit